crates/viewer/re_view_spatial/tests/transform_clamping.rs

```diff
@@ -0,0 +1,261 @@
+    })
+}
+
+fn run_view_ui_and_save_snapshot(
+    test_context: &mut TestContext,
+    (view_id_boxes, view_id_spheres): (ViewId, ViewId),
+    name: &str,
+    size: egui::Vec2,
+) {
+    for (target, view_id) in [("boxes", view_id_boxes), ("spheres", view_id_spheres)] {
+        let mut harness = test_context
+            .setup_kittest_for_rendering()
+            .with_size(size)
+            .build(|ctx| {
+                re_ui::apply_style_and_install_loaders(ctx);
+
+                egui::CentralPanel::default().show(ctx, |ui| {
+                    test_context.run(ctx, |ctx| {
+                        let view_class = ctx
+                            .view_class_registry
+                            .get_class_or_log_error(SpatialView3D::identifier());
+
+                        let view_blueprint = ViewBlueprint::try_from_db(
+                            view_id,
+                            ctx.store_context.blueprint,
+                            ctx.blueprint_query,
+                        )
+                        .expect("we just created that view");
+
+                        let mut view_states = test_context.view_states.lock();
+
+                        let view_state = view_states.get_mut_or_create(view_id, view_class);
+                        let (view_query, system_execution_output) =
+                            re_viewport::execute_systems_for_view(
+                                ctx,
+                                &view_blueprint,
+                                ctx.current_query().at(), // TODO(andreas): why is this even needed to be passed in?
+                                view_state,
+                            );
+                        view_class
+                            .ui(ctx, ui, view_state, &view_query, system_execution_output)
+                            .expect("failed to run view ui");
+                    });
+
+                    test_context.handle_system_commands();
+                });
+            });
+
+        {
+            // This test checks the clamping behavior of components & instance poses on boxes & spheres.
+            //
+            // One view shows spheres, the other boxes.
+            //
+            // For both you should see:
+            // * 2x red (one bigger than the other)
+            // * 1x green
+            // * 2x blue (one bigger than the other)
+            // * NO other boxes/spheres, in particular no magenta ones!
+
+            let name = format!("{name}_{target}");
+            let raw_input = harness.input_mut();
+            raw_input
+                .events
+                .push(egui::Event::PointerMoved((150.0, 150.0).into()));
+            raw_input.events.push(egui::Event::MouseWheel {
+                unit: egui::MouseWheelUnit::Line,
+                delta: egui::Vec2::UP * 2.0,
+                modifiers: egui::Modifiers::default(),
+            });
+            harness.run_steps(10);
+            harness.snapshot(&name);
```

Comment:
We'll probably run into the same img comparision issues here again as on the draw order test, ci will tell
if the threshhold seems high to you (don't ask what that means ;)) please link the re_renderer antialiasing ticket